### ISSUE TITLE
Translators should be able to gain rank progress for the translated language regardless of whether he/she has written previous translations to that Kata or edited it through the Kata Editor

### ISSUE THREAD

**Issue by DonaldKellett** [codewars/codewars.com#1163]
Today I just translated ["Battleship Field Validator" (`3 kyu`)](https://www.codewars.com/kata/battleship-field-validator) into C++ which was approved.  I got all 64 honor for translating a blue Kata; however, my rank progress in C++ stayed right at `5 kyu / 72%` even after an hour or so since the translation was approved.

Initially I thought that the CW servers were perhaps a bit slow in updating my rank progress; however, I then went to the "Solutions" tab of the Haskell/Kotlin version of a few `3 kyu` Kata that I have previously translated into PHP but **not** completed in Haskell/Kotlin.  As I _partially_ anticipated, the Haskell/Kotlin solutions were displayed to me immediately without warning of forfeiting the Kata in those languages which implied that I have forfeited rank progress in all future languages versions of this Kata.  That's when I remembered that submitting an approved translation to a Kata sometimes counted as a Kata Edit which would trigger this "complete forfeiture" response.

I personally think that it is OK if a user forfeits eligibility to all other **currently published** language versions of a Kata once he/she has submitted a translation to that Kata which gets approved (because he/she could view the other translations in the process which could give him/her an unfair edge in those languages).  However, I do not understand why submitting an approved translation in one language (or even having gone to the Kata Editor once) should cause the user to completely forfeit all **future language versions** to that Kata?  Specifically, if a user submits an approved translation to a language (and maybe later then even edits the Kata to fix a few bugs in certain language versions, for example) and then submits _another_ approved translation to another language for that Kata **in the future**, shouldn't he/she gain individual rank progress for **both** languages?  Point being, the Kata translator couldn't've possibly gained an unfair edge in the language that he/she translated the Kata into.

I hope I expressed my point clearly; feel free to ping me if you didn't quite understand my point.  What do you think jhoffner ?

┆Issue is synchronized with this [Clickup ](https://app.clickup.com/t/2me1bd7) by [Unito](https://www.unito.io)

**Comment by javatlacati** [comment]
I think this is the way it behaves now, can you verify @DonaldKellett ?

**Comment by Voileexperiments** [comment]
I think it does now. Because otherwise I shouldn't be getting almost `1dan` in Ruby and already `1dan` in Python, while my Haskell rank is still early `1kyu`. I've solved more purple Haskell katas than purple Python/Ruby katas.

**Comment by anter69** [comment]
I started translating katas lately, and I experience the same behavior as described above, in the following situations:
* I solve a kata, then edit it because there is something I can fix, and then submit a translation which is later approved;
* or I forfeit a kata because I  don't know that language (but I can easily reproduce the kata in my languages), then I submit a translation which is later approved,

...and I don't seem to get any honor or % in the translated language. Although it's not a big deal, it still spoils the "game" a bit.

**Comment by Blind4Basics** [comment]
Yes, as soon as you open the edit panel, you lose all occasions to get progress in any language.

**Comment by FArekkusu** [comment]
Turns out there's an inconsistency in the "losing progress eligibility through translating a kata" rule:
* You **do** lose eligibility in the new languages whoever the translator is
* But you **do not** lose eligibility in the old languages which had been available before you translated the kata.

**Comment by Blind4Basics** [comment]
>   * You **do** lose eligibility in the new languages which become available after you've translated the kata.
> 
>   * But you **do not** lose eligibility in the old languages which were available before you translated the kata.

Yep, confirmed, and that doesn't make any sense! x)

Just now, I opened the edit panel of a kata I approved recently, went to the JS version in there, went then to the JS solutions page, which was still locked (tho I'm a contributor...). I even ""completed""  the kata in JS. That triggered honor gain, (tho since I didn't get them after the approval for python, that's not a problem. See https://github.com/Codewars/codewars.com/issues/1783). Unfortunately, I don't remember how many % I had before in JS so I cannot tell if I got them too. :o

Anyway, the point is: going to the edit panel should discard all existing languages (one can complete those before going to the edit panel), but shouldn't discard eligibility for languages published after, not the other way around, like it seems to be the case right now.
